src/VisualStudio/Core/Def/Implementation/Progression/GraphBuilder.cs

```diff
@@ -690,6 +691,98 @@ public GraphNode AddNodeForDocument(Document document)
+                NavigateToItemKind.Structure => CodeNodeCategories.Struct,
+                NavigateToItemKind.Method => CodeNodeCategories.Method,
+                NavigateToItemKind.Property => CodeNodeCategories.Property,
+                NavigateToItemKind.Event => CodeNodeCategories.Event,
+                NavigateToItemKind.Constant or
+                NavigateToItemKind.EnumItem or
+                NavigateToItemKind.Field => CodeNodeCategories.Field,
+                _ => null,
+            };
+
+            // If it's not a category that progression understands, then ignore.
+            if (category == null)
+                return null;
+
+            // Get or make a node for this symbol's containing document that will act as the parent node in the UI.
+            var documentNode = this.AddNodeForDocument(document);
+
+            // For purposes of keying this node, just use the display text we will show.  In practice, outside of error
+            // scenarios this will be unique and suitable as an ID (esp. as these names are joined with their parent
+            // document name to form the full ID).
+            var label = result.NavigableItem.DisplayTaggedParts.JoinText();
+            var id = documentNode.Id.Add(GraphNodeId.GetLiteral(label));
+
+            // If we already have a node that matches this (say there are multiple identical sibling symbols in an error
+            // situation).  We just ignore the second match.
+            var existing = _graph.Nodes.Get(id);
+            if (existing != null)
+                return null;
+
+            var symbolNode = _graph.Nodes.GetOrCreate(id);
+
+            symbolNode.Label = label;
+            symbolNode.AddCategory(category);
+            symbolNode[DgmlNodeProperties.Icon] = GetIconString(result.NavigableItem.Glyph);
+            symbolNode[RoslynGraphProperties.ContextDocumentId] = document.Id;
+            symbolNode[RoslynGraphProperties.ContextProjectId] = project.Id;
+
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var span = text.Lines.GetLinePositionSpan(NavigateToUtilities.GetBoundedSpan(result.NavigableItem, text));
+
+            symbolNode[CodeNodeProperties.SourceLocation] = new SourceLocation(
+                document.FilePath,
+                new Position(span.Start.Line, span.Start.Character),
+                new Position(span.End.Line, span.End.Character));
+
+            this.AddLink(documentNode, GraphCommonSchema.Contains, symbolNode);
+
+            return symbolNode;
+        }
+
+        private static string GetIconString(Glyph glyph)
+        {
+            var groupName = glyph switch
+            {
+                Glyph.ClassPublic or Glyph.ClassProtected or Glyph.ClassPrivate or Glyph.ClassInternal => "Class",
+                Glyph.ConstantPublic or Glyph.ConstantProtected or Glyph.ConstantPrivate or Glyph.ConstantInternal => "Field",
```

Comment:
Do we use the same glyph for constants and fields?

Reply:
Yup!  As per GetOrCreateNodeForFieldAsync it looks like we only distinguish fields and enum-members in progression.